File: internal/db/model/post.go
package model

import (
	"database/sql"
	"time"
)

// Post 豆瓣帖子模型w
type Post struct {
	ID          int64                  `db:"id"`
	PostID      string                 `db:"post_id"`
	GroupID     string                 `db:"group_id"`
	AuthorInfo  map[string]interface{} `db:"author_info"` // 内存中为结构化类型，DB 中以 JSON 字符串存储
	Alt         string                 `db:"alt"`
	Title       string                 `db:"title"`
	Content     string                 `db:"content"`
	PhotoList   []string               `db:"photo_list"`  // 内存为切片，DB 中以 JSON 存储
	Rent        sql.NullFloat64        `db:"rent"`
	Subway      sql.NullString         `db:"subway"`
	Contact     sql.NullString         `db:"contact"`
	IsMatched   bool                   `db:"is_matched"`
	KeywordList []string               `db:"keyword_list"` // 内存为切片，DB 中以 JSON 存储
	Comment     sql.NullString         `db:"comment"`
	IsCollected bool                   `db:"is_collected"`
	Created     time.Time              `db:"created"`
	Updated     time.Time              `db:"updated"`
	CreatedAt   time.Time              `db:"created_at"`
	UpdatedAt   sql.NullTime           `db:"updated_at"`
}

// TableName Post表名
func (p *Post) TableName() string {
	return "Post"
}

File: src/main/java/com/douban/bot/model/Post.java
package com.douban.bot.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;
import java.util.List;
import java.util.Map;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Post {
    private Long id;
    private String postId;
    private String groupId;
    private Map<String, Object> authorInfo;
    private String alt;
    private String title;
    private String content;
    private List<String> photoList;
    private Double rent;
    private String subway;
    private String contact;
    private Boolean isMatched;
    private List<String> keywordList;
    private String comment;
    private Boolean isCollected;
    private Boolean botReplied;  // 是否已自动回复
    private String botReplyContent;  // 自动回复内容
    private LocalDateTime botReplyAt;  // 自动回复时间
    private LocalDateTime created;
    private LocalDateTime updated;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
}
